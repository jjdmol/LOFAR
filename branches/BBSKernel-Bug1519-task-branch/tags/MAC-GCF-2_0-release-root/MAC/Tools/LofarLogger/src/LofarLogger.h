//#  LofarLogger.h: logger wrapper
//#
//#  Copyright (C) 2002-2003
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id$

#ifndef LOFAR_LOGGER_H
#define LOFAR_LOGGER_H

#include <Common/lofar_string.h>
#include <Common/lofar_map.h>
#include <log4cplus/logger.h>
#include "LofarLoggerNames.h"

// forward declaration

// define convenience macros
#define LOFAR_LOG_TRACE_SCOPE(logger,logEvent) \
    LofarScopeTraceLogger _scopeLogger(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_TRACE(logger,logEvent) \
    LofarLogger::getInstance().trace(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_DEBUG(logger,logEvent) \
    LofarLogger::getInstance().debug(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_INFO(logger,logEvent)  \
    LofarLogger::getInstance().info(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_WARN(logger,logEvent)  \
    LofarLogger::getInstance().warn(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_ERROR(logger,logEvent) \
    LofarLogger::getInstance().error(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);
#define LOFAR_LOG_FATAL(logger,logEvent) \
    LofarLogger::getInstance().fatal(logger,LofarLogger::formatString logEvent ,__FILE__,__LINE__);

/**
 *
 * The LofarLogger class wraps the log4cplus::Logger class
 *
 * Usage:
 *
 *    #define EXAMPLE_LOGGER_NAME "ASTRON.LOFAR.MAC.ExampleLoggerName"
 *
 *    LOFAR_LOG_TRACE(EXAMPLE_LOGGER_NAME,("example %d level message %d","trace",15))
 *    LOFAR_LOG_FATAL(EXAMPLE_LOGGER_NAME,("example fatal level message"))
 *
 * NOTE: the second argument of the macro HAS to be surrounded with brackets.
 */
class LofarLogger
{
    public:

        /**
        * destructor
        */
        virtual ~LofarLogger();

        static LofarLogger getInstance();

        /**
        * logging methods4
        */
        void trace(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);
        void debug(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);
        void info(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);
        void warn(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);
        void error(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);
        void fatal(const string& logger,const string& logEvent,const char* file=NULL,int line=-1);

        log4cplus::Logger& getLogger(const string& logger);
        static string formatString(char* format,...);

    protected:
        /**4
        * constructor. Protected because public construction is not allowed
        */
        LofarLogger();
        LofarLogger(const LofarLogger&);

    private:

        /**
        * Don't allow copying of the LofarLogger object.
        */
        LofarLogger& operator=(const LofarLogger&);

        typedef std::map<const std::string,log4cplus::Logger> TLoggerMap;

        /**
        * keep the loggers in a map for re-use
        */
        TLoggerMap         _loggerMap;
        static LofarLogger _sLofarLogger;
};

/**
 *
 * The LofarScopeTraceLogger class logs trace messages in its constructor and destructor
 *
 */
class LofarScopeTraceLogger
{
    public:
       explicit LofarScopeTraceLogger(const std::string& logger, const std::string& logEvent,const char* file=NULL,int line=-1) :
                    _traceLogger(LofarLogger::getInstance().getLogger(logger),logEvent,file,line)
                {
                };
                ~LofarScopeTraceLogger()
                {
                };

    protected:
                /**
                * constructor. Protected because public construction is not allowed
                */
                LofarScopeTraceLogger();
                LofarScopeTraceLogger(const LofarScopeTraceLogger&);
                LofarScopeTraceLogger& operator=(const LofarScopeTraceLogger&);

    private:
                log4cplus::TraceLogger _traceLogger;
};

#endif
