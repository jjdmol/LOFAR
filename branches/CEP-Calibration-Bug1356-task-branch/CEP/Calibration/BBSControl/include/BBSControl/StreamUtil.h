//#  StreamUtil.h: Specializations of the <Common/StreamUtil.h>.
//#
//#  Copyright (C) 2002-2008
//#  ASTRON (Netherlands Foundation for Research in Astronomy)
//#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
//#
//#  This program is free software; you can redistribute it and/or modify
//#  it under the terms of the GNU General Public License as published by
//#  the Free Software Foundation; either version 2 of the License, or
//#  (at your option) any later version.
//#
//#  This program is distributed in the hope that it will be useful,
//#  but WITHOUT ANY WARRANTY; without even the implied warranty of
//#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//#  GNU General Public License for more details.
//#
//#  You should have received a copy of the GNU General Public License
//#  along with this program; if not, write to the Free Software
//#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//#
//#  $Id$

#ifndef LOFAR_BBSCONTROL_STREAMUTIL_H
#define LOFAR_BBSCONTROL_STREAMUTIL_H

// \file
// Useful stream manipulation methods.

//# Includes
#include <Common/StreamUtil.h>

namespace LOFAR
{
  // \ingroup BBSControl
  // @{

  // Specialization of writeVector for vector of string. We need this, because
  // we want to enclose each string with quotes.
  template<>
  void writeVector (ostream& os, const vector<string>& vec,
                    const char* separator,
                    const char* prefix, const char* postfix);

  template<typename T>
  string toString(const vector<T>& vec)
  {
    ostringstream oss;
    oss << vec;
    return oss.str();
  }

  // @}

} // namespace LOFAR

#endif
