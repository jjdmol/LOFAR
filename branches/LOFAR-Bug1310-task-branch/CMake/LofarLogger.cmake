#  LofarLogger.cmake: Check for logger (log4cplus, log4cxx or none)
#
#  Copyright (C) 2008-2009
#  ASTRON (Netherlands Foundation for Research in Astronomy)
#  P.O.Box 2, 7990 AA Dwingeloo, The Netherlands, seg@astron.nl
#
#  This program is free software; you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation; either version 2 of the License, or
#  (at your option) any later version.
#
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with this program; if not, write to the Free Software
#  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
#
#  $Id$

include(LofarFindPackage)

macro(lofar_logger)

  if(USE_LOG4CPLUS AND USE_LOG4CXX)
    message(FATAL_ERROR "Cannot use Log4Cplus and Log4Cxx at the same time")
  endif(USE_LOG4CPLUS AND USE_LOG4CXX)

  if(USE_LOG4CPLUS)
    MESSAGE(STATUS "Checking for Log4Cplus ...")
    lofar_find_package(Log4Cplus 1 log4cplus/logger.h "log4cplus jpeg")
  endif(USE_LOG4CPLUS)

  if(USE_LOG4CXX)
    MESSAGE(STATUS "Checking for Log4Cxx ...")
    lofar_find_package(Log4Cxx 1 log4cxx/logger.h)
  endif(USE_LOG4CXX)

endmacro(lofar_logger)
