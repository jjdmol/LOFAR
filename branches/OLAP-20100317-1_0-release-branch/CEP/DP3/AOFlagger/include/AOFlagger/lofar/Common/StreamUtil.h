//# StreamUtil.h: useful stream manipulation methods.
//#
//# Copyright (C) 2002-2003
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id: StreamUtil.h 14057 2009-09-18 12:26:29Z diepen $

#ifndef LOFAR_COMMON_STREAMUTIL_H
#define LOFAR_COMMON_STREAMUTIL_H

// \file
// Useful stream manipulation methods.

//# Includes
#include <Common/LofarTypes.h>
#include <Common/lofar_string.h>
#include <Common/lofar_vector.h>
#include <Common/lofar_iosfwd.h>

namespace LOFAR
{
  // Handles indentation of text lines. Every time an Indent object is
  // constructed, the static member \c level is incremented by one, and every
  // time an Indent object is destructed \c level is decremented by one. To
  // increment the amount of indentation you simply create an Indent
  // object. When this object goes out of scope, the amount of indentation is
  // automagically decremented.
  class Indent {
  public:
    // Constructor. Increments indentation level.
    Indent() { lvl++; }
    // Destructor. Decrements indentation level.
    ~Indent() { lvl--; }
    // Return the amount of indentation.
    static uint level() { return lvl; }
    // Return the token to be printed per indentation level
    static const string& token() { return tok; }
  private:
    // Indentation level.
    static uint lvl;
    // Token to be printed per indentation level.
    static const string tok;
  };


  // Write a vector to an ostream with a given separator, prefix and postfix.
  template<class T>
  void writeVector (std::ostream& os, const std::vector<T>& vec,
		    const char* separator=",",
		    const char* prefix="[", const char* postfix="]")
  {
    os << prefix;
    for (uint i = 0; i < vec.size(); i++) {
      if (i > 0) os << separator;
      os << vec[i];
    }
    os << postfix;
  }


  // Print an indentation that depends on the number of Indent objects
  // currently in existence.
  inline ostream& indent(ostream& os) 
  {
    for (uint i = 0; i < Indent::level(); ++i) {
      os << Indent::token();
    }
    return os;
  }


  // Print the contents of a vector enclosed in square brackets, using a comma
  // as separator.
  // \note operator<<() must be defined for type \c T.
  template<typename T>
  inline ostream& operator<<(ostream& os, const vector<T>& v)
  {
    writeVector(os, v, ",", "[", "]");
    return os;
  }


} // namespace LOFAR

#endif
