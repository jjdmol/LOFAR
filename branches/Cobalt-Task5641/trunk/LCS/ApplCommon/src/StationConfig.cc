//# StationConfig.cc: Interface to the RemoteStation.conf file.
//#
//# Copyright (C) 2009
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id$

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>

//# Includes
#include <Common/LofarLogger.h>
#include <Common/ParameterSet.h>
#include <ApplCommon/StationConfig.h>

namespace LOFAR {

//---------- global instance of StationConfig ----------
static StationConfig* globalStationConfigInstance = 0;

StationConfig* globalStationConfig()
{
	if (globalStationConfigInstance == 0) {
		globalStationConfigInstance = new StationConfig();
	}
	return (globalStationConfigInstance);
}

//
// Constructor
//
StationConfig::StationConfig()
{
	// Try to find the configurationfile
	ConfigLocator   CL;
	string          fileName(CL.locate("RemoteStation.conf"));
	ASSERTSTR(!fileName.empty(), "Cannot find the station configurationfile 'RemoteStation.conf'");

	ParameterSet    StationInfo(fileName);
	stationID	 = StationInfo.getInt ("RS.STATION_ID");
	nrRSPs		 = StationInfo.getInt ("RS.N_RSPBOARDS");
	nrTBBs		 = StationInfo.getInt ("RS.N_TBBOARDS");
	nrLBAs		 = StationInfo.getInt ("RS.N_LBAS");
	nrHBAs		 = StationInfo.getInt ("RS.N_HBAS");
	hasSplitters = StationInfo.getBool("RS.HBA_SPLIT");
	hasAartfaac  = StationInfo.getBool("RS.AARTFAAC");
	hasWideLBAs	 = StationInfo.getBool("RS.WIDE_LBAS");

	LOG_DEBUG(formatString("Stations has %d LBA and %d HBA antennas and %ssplitters",
											nrLBAs, nrHBAs, (hasSplitters ? "" : "NO ")));
}

StationConfig::~StationConfig()
{ }

} // namespace LOFAR
