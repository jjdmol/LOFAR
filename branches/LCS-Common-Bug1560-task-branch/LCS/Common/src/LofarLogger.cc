//# LofarLogger.cc: Interface to the log4cplus logging package
//#
//# Copyright (C) 2004
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id$

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>

//# Includes
#include <cstdio>					// snprintf
#include <unistd.h>					// readlink
#include <libgen.h>				// basename
#include <cstring>
#include <Common/LofarLogger.h>

#if defined(HAVE_LOG4CXX)
# include <log4cxx/ndc.h>
#endif


namespace LOFAR {

//#------------------------- Internal implementation -----------------------------
//
// lofarLoggerInitNode()
//
// Creates a NDC with the text "application@node" and pushes it
// on the NDC stack
void lofarLoggerInitNode(void) {
	const int	MAXLEN = 128;
	int		applNameLen = 0;
	char	hostName [MAXLEN];
	char	applName [MAXLEN];
	char	loggerId [MAXLEN];

	// try to resolve the hostname
	if (gethostname (hostName, MAXLEN-1) < 0) {
		hostName[0]='\0';
	}

	// try to resolve the applicationname
	applNameLen = readlink("/proc/self/exe", applName, MAXLEN-1);
	if (applNameLen >= 0)
		applName[applNameLen] = '\0';
	else
		strcpy (applName, "");
//	}	

	// construct loggerId and register it.
	snprintf(loggerId, MAXLEN-1, "%s@%s", basename(applName), hostName);
#if defined(HAVE_LOG4CPLUS)
	log4cplus::getNDC().push(loggerId);
#elif defined(HAVE_LOG4CXX)
        log4cxx::NDC::push(loggerId);
#endif
}

}	// namespace LOFAR
