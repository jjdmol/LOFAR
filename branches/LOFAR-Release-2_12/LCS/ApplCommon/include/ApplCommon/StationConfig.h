//# StationConfig.h: Interface class for accessing RemoteStation.conf
//#
//# Copyright (C) 2009
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id$

#ifndef LOFAR_APPLCOMMON_STATION_CONFIG_H
#define LOFAR_APPLCOMMON_STATION_CONFIG_H

// \file
// Interface class for accessing RemoteStation.conf

//# Never #include <config.h> or #include <lofar_config.h> in a header file!

//# Includes
#include <Common/lofar_fstream.h>
#include <Common/LofarLogger.h>
#include <Common/LofarLocators.h>

namespace LOFAR {

// \addtogroup ApplCommon
// @{

//# Forward Declarations
//class forward;

// The StationConfig class is an interface for the RemoteStation.conf file.
// It reads in the file and stores the values in its datamembers. All datamembers
// are public for easy access.

class StationConfig
{
public:
	StationConfig();
	~StationConfig();

private:
	// Copying is not allowed.
	StationConfig (const StationConfig& that);
	StationConfig& operator= (const StationConfig& that);

public:
	//# Data members
	int		stationID;
	int		nrRSPs;
	int		nrTBBs;
	int		nrLBAs;
	int		nrHBAs;
	bool	hasSplitters;
	bool	hasAartfaac;
	bool	hasWideLBAs;
};

// Make one instance of the Station configuration globally accessable.
StationConfig*	globalStationConfig();

// @}

} // namespace LOFAR

#endif
