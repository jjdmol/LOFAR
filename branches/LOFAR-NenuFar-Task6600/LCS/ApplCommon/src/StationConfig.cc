//# StationConfig.cc: Interface to the RemoteStation.conf file.
//#
//# Copyright (C) 2009
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id$

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>

//# Includes
#include <Common/LofarLogger.h>
#include <Common/ParameterSet.h>
#include <Common/StreamUtil.h>
#include <ApplCommon/StationConfig.h>

namespace LOFAR {

//---------- global instance of StationConfig ----------
static StationConfig* globalStationConfigInstance = 0;

StationConfig* globalStationConfig()
{
	if (globalStationConfigInstance == 0) {
		globalStationConfigInstance = new StationConfig();
	}
	return (globalStationConfigInstance);
}

//
// Constructor
//
StationConfig::StationConfig(const string& filename)
{
	// Try to find the configurationfile
	ConfigLocator   CL;
	string          fileName(CL.locate(filename));
	ASSERTSTR(!fileName.empty(), "Cannot find the station configurationfile '" << filename << "'");

	ParameterSet    StationInfo(fileName);
	stationID	 = StationInfo.getInt ("RS.STATION_ID");
	nrRSPs		 = StationInfo.getInt ("RS.N_RSPBOARDS");
	nrTBBs		 = StationInfo.getInt ("RS.N_TBBOARDS");
	hasSplitters = StationInfo.getBool("RS.HBA_SPLIT");
	hasAartfaac  = StationInfo.getBool("RS.AARTFAAC");
	hasWideLBAs	 = StationInfo.getBool("RS.WIDE_LBAS");
	if (StationInfo.isDefined("RS.ANT_TYPES")) {
		antTypes = StationInfo.getStringVector("RS.ANT_TYPES");
	}
	else {
		antTypes.push_back("LBA");
		antTypes.push_back("HBA");
	}
	nrTypes = antTypes.size();
	for (int i = 0; i < nrTypes; ++i) {
		antCounts.push_back(StationInfo.getInt(formatString("RS.N_%sS", antTypes[i].c_str())));
	}
	LOG_DEBUG_STR("AntennaFields: " << antTypes << ", counts: " << antCounts << 
			formatString(" and %ssplitters", (hasSplitters ? "" : "NO ")));

}

StationConfig::~StationConfig()
{ }

int StationConfig::nrAntennas(const string& antTypeName) const
{
	for (size_t i = 0; i < antTypes.size(); ++i) {
		if (antTypes[i] == antTypeName) {
			return (antCounts[i]);
		}
	}
	return (0);
}

} // namespace LOFAR
