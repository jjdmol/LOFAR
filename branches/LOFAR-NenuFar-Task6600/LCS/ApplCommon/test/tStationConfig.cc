//# tStationConfig.cc
//#
//# Copyright (C) 2002-2004
//# ASTRON (Netherlands Institute for Radio Astronomy)
//# P.O.Box 2, 7990 AA Dwingeloo, The Netherlands
//#
//# This file is part of the LOFAR software suite.
//# The LOFAR software suite is free software: you can redistribute it and/or
//# modify it under the terms of the GNU General Public License as published
//# by the Free Software Foundation, either version 3 of the License, or
//# (at your option) any later version.
//#
//# The LOFAR software suite is distributed in the hope that it will be useful,
//# but WITHOUT ANY WARRANTY; without even the implied warranty of
//# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
//# GNU General Public License for more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with the LOFAR software suite. If not, see <http://www.gnu.org/licenses/>.
//#
//# $Id: tStationConfig.cc 22376 2012-10-17 10:00:37Z mol $

//# Always #include <lofar_config.h> first!
#include <lofar_config.h>

//# Includes
#include <Common/LofarLogger.h>
#include <Common/StreamUtil.h>
#include <ApplCommon/StationConfig.h>

using namespace LOFAR;

void show (const StationConfig&	SC)
{
	cout << "StationID    = " << SC.stationID << endl;
	cout << "nr RSPboards = " << SC.nrRSPs << endl;
	cout << "nr TBboards  = " << SC.nrTBBs << endl;
	cout << "HBA split    = " << SC.hasSplitters << endl;
	cout << "wide LBAs    = " << SC.hasWideLBAs << endl;
	cout << "Aartfaac     = " << SC.hasAartfaac << endl;
	cout << "nr antenna types = " << SC.nrTypes << endl;
	cout << "antenna types= " << SC.antTypes << endl;
	for (int i = 0; i < SC.nrTypes; i++) {
		cout << "field " << SC.antTypes[i] << ": " << SC.antCounts[i] << endl;
		ASSERTSTR(SC.antCounts[i] == SC.nrAntennas(SC.antTypes[i]), "Illegal count returned for nrAntennas(" << SC.antTypes[i] << ")");
	}
}

int main (int/*argc*/, char* argv[]) 
{
	INIT_LOGGER(argv[0]);

	StationConfig	oldConfig("tStationConfig.in_oldFormat");
	show(oldConfig);
	
	StationConfig	newConfig("tStationConfig.in_newFormat");
	show(newConfig);
	
	try {
		StationConfig	wrongConfig("tStationConfig.in_wrongFormat");
	} catch (LOFAR::APSException&	ex) {
	}
	
	return (0);
}
