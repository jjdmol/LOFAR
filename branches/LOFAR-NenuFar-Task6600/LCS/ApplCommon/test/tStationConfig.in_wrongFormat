#
# THIS FILE IS GENERATED, DO NOT MODIFY IT.
#
# RemoteStation.conf for CS001
#
# Describes the amount of available hardware on the station.
#
RS.STATION_ID  = 1
RS.N_RSPBOARDS = 24
RS.N_TBBOARDS  = 12
RS.FIELDS      = [LBA, HBA, NNF]
RS.N_LBAS      = 96
RS.N_HBAS      = 96
RS.HBA_SPLIT   = Yes
RS.WIDE_LBAS   = Yes
RS.AARTFAAC    = No

