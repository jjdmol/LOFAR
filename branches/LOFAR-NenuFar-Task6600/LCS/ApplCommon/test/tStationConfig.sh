#!/bin/sh
./runctest.sh tStationConfig
