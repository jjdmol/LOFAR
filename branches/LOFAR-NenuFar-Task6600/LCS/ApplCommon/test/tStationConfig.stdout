StationID    = 1
nr RSPboards = 24
nr TBboards  = 12
HBA split    = 1
wide LBAs    = 1
Aartfaac     = 0
nr antenna types = 2
antenna types= [LBA,HBA]
field LBA: 96
field HBA: 96
StationID    = 1
nr RSPboards = 24
nr TBboards  = 12
HBA split    = 1
wide LBAs    = 1
Aartfaac     = 0
nr antenna types = 3
antenna types= [LBA,HBA,NNF]
field LBA: 96
field HBA: 96
field NNF: 17
